packages/core/src/snapshot.js

```diff
@@ -86,8 +88,22 @@ function mapSnapshotOptions(snapshots, context) {
     // transform snapshot URL shorthand into an object
     if (typeof snapshot === 'string') snapshot = { url: snapshot };
 
-    // normalize the snapshot url and use it for the default name
+    try {
+      // encoding snapshot url, if contians invalid URI characters/syntax
+      let modifiedURL = modifyURL(snapshot.url);
```

Comment:
Above code and this code is exactly same except the URL. Don't duplicate the code

Reply:
- The reason to handle try catch separately, is because if we log the error in utils.js, we will not know from where we are logging this error either from snapshot or network etc. So Because of that parent function needs to handle it.

However, to avoid here, I am also passing `logger` now, from the parent function to log the error. This will ensure, from where it is getting logged.